src/IntelliHR/Validation/Validators/MaxDate.php

```diff
@@ -2,65 +2,51 @@
 
 namespace IntelliHR\Validation\Validators;
 
-use DateTime;
+use Exception;
 use Illuminate\Contracts\Validation\Validator;
-use InvalidArgumentException;
 
 class MaxDate extends AbstractValidator
 {
     /**
-     * Name of the validator
-     *
      * @var string
      */
     public static $name = 'max_date';
 
     /**
-     * Fallback message
-     *
      * @var string
      */
     public static $message = ':attribute must be before :max_date';
 
-    /**
-     * @param           $attribute
-     * @param           $value
-     * @param array     $parameters
-     * @param Validator $validator
-     *
-     * @return bool
-     * @throws InvalidArgumentException
-     */
     public function validateMaxDate(
         $attribute,
         $value,
         array $parameters,
         Validator $validator
-    ) {
+    ): bool {
         $this->requireParameterCount(1, $parameters, self::$name);
+        $format = $parameters[1] ?? null;
+
+        $maxDate = $this->getDateForParameter($parameters[0], $format);
+
+        try {
+            $date = $this->getDateForParameter($value, $format);
+        } catch (Exception $e) {
+            return true;
+        }
 
-        $maxDate = new DateTime($parameters[0]);
-        $date = new DateTime($value);
+        if ($date === false || $maxDate === false) {
+            return true;
+        }
 
         return ($date <= $maxDate);
```

Comment:
Since you're making a lot of changes around here, I'd replace the usage of `DateTime` objects with `Carbon` objects instead and then use `$date->lte($maxDate)`. What is actually happening here is the `<=` is converting the 2 dates to a string and then comparing them as string values.